Point-of-No-Return/Collision.h

```diff
@@ -0,0 +1,63 @@
+﻿#ifndef COLLISION_H_
+#define COLLISION_H_
+
+#include "Character.h"
+#include "Mapchip.h"
+
+/**
+* @brief 当たり判定をする関数
+*/
+namespace Collision
+{
+	/**
+	 * @brief 主人公周りのブロックで当たり判定があるかどうか探す
+	 * @param  vec  キャラクターの位置
+	 * @param  size キャラクターのサイズ
+	 * @param  map  マップの情報のアドレス
+	 * @return 当たり判定をとるブロックの座標の集まり
+	 */
+	std::vector<Vec2> SearchBlock(const Vec2& vec, const Size& size, int** map);
+
+	// TODO: collision引数Hero修正しないといけない
+	/**
+	 * @brief ブロックに当たっているかどうかの確認関数
+	 * @param hero     主人公の情報
+	 * @param previous キャラクターの前の位置
+	 * @param size     キャラクターのサイズ
+	 * @param vec      キャラクターの移動量や方向
+	 * @param vec2     当たり判定をとるブロックの座標の集まり
+	 */
+	void CheckBlock(Hero& hero, Vec2 previous, Size size, Vec2 vec, std::vector<Vec2> vec2);
+
+	// TODO: collision引数Hero修正しないといけない
+	/**
+	 * @brief どの辺で当たっているのかの確認関数
+	 * @param hero     主人公の情報
+	 * @param previous キャラクターの前の位置
+	 * @param size     キャラクターのサイズ
+	 * @param vector   キャラクターの移動量や方向
+	 * @param vec2     当たり判定をとるブロックの座標の集まり
+	 * @param i        当たり判定をとるブロックを判断する
+	 */
+	void HitCheckEdge(Hero& hero, Vec2 previous, Size size, Vec2 vector, std::vector<Vec2> vec2, int i);
+
+	inline bool operator < (const CoordinateX& valu,const float& value)
```

Comment:
```suggestion
	inline bool operator < (const CoordinateX& lhs, const float& rhs)
```
とかの方が良いかと。
https://toburau.hatenablog.jp/entry/20090524/1243177092

あと、 `<` の後のスペースは不要では？という話は大浦にしたと思います。

Reply:
> あと、 < の後のスペースは不要では？という話は大浦にしたと思います。

[このコメント](https://github.com/human-osaka-game-2019/Point-of-No-Return/pull/41#discussion_r344446164) ですね。